### ISSUE TITLE
Bug: [array-type] Does not validate shape of config

### ISSUE THREAD

**Issue by domdomegg** [typescript-eslint/typescript-eslint#6892]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.3&sourceType=module&code=MYewdgzgLgBAtgTwMpQE4EswHMIC4bQbYDaAujALwxkDcAUKJLCFABYCmqKRO+AgqlQBDBAB5CmLAD5K1UvSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AQ2mncNqNPRgA2pERcA9tEgAaKJ26FIAXXBgAviDVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const myStrings: string[] = [];
const otherStrings: Array<string> = [];
```
```


### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/array-type": ["error", "array"]
  },
};
```

### Expected Result

I initially expected that I'd enforce array types (e.g. `string[]` rather than `Array<string>` everywhere), becuase I skimmed https://github.com/typescript-eslint/typescript-eslint/blob/HEAD/packages/eslint-plugin/docs/rules/array-type.md and didn't realise this should be an options object.

This is a user error, and should instead be configured:

```
    "@typescript-eslint/array-type": ["error", { default: "array" }]
```

That said, I then expected the rule would give me some kind of warning or error that my eslint config was invalid, so I could debug this quickly.

### Actual Result

The rule doesn't complain about its configuration, but instead highlights every array type it sees as an error - and tries to convert them to the other type in a loop.

### Additional Info

There is also https://github.com/typescript-eslint/typescript-eslint/issues/6852, which relates to other config not being validated by this rule.

**Comment by bradzacher** [comment]
This is a duplicate of #6852, yeah - exact same problem.
